src/XdebugHandler.php

```diff
@@ -260,7 +260,24 @@ private function doRestart($command)
         $this->tryEnableSignals();
         $this->notify(Status::RESTARTING, $command);
 
-        passthru($command, $exitCode);
+        // Use proc_open on linux/OSX and on recent Windows where stream_isatty may be used
+        // This is to ensure the process pipes remain bound to a tty which does not happen with passthru
+        if (function_exists('proc_open') && (!\defined('PHP_WINDOWS_VERSION_BUILD') || function_exists('stream_isatty'))) {
```

Comment:
What was wrong with just checking for the function rather than stream_isatty (which will either exist or not, regardless of the OS)?

Reply:
The point here is that we need to make sure a few things are present:

- proc_open needs to be callable (it's probably typically disabled in silly shared hosts)
- if not windows, just use it because it'll most likely work and there we will check using posix_isatty on older PHP, so we need to make sure the pipes are set correctly
- if windows, we only need proc_open if stream_isatty is present, because otherwise composer will not check anything and assume the pipe is a tty/interactive. AFAIK posix_isatty does not exist on windows? If that is wrong then we always need to run proc_open there too and the whole `&& ...` bit can be dropped.

Reply:
Sure, but you can use fstat on builds without stream_isatty or posix_isatty. I would have thought just checking that the function exists is fine.

Reply:
Possibly.. that'd need fixing in Composer tho then as we only rely on `*_isatty` atm. Anyway feel free to simplify the code here :)

Reply:
PHP itself uses fstat as a falback in stream_isatty (just to make you feel happier!)